brave/core/permission/model.py

```
@@ -0,0 +1,88 @@
+# encoding: utf-8
+
+from __future__ import unicode_literals
+
+from datetime import datetime
+from mongoengine import Document, EmbeddedDocument, EmbeddedDocumentField, StringField, EmailField, URLField, DateTimeField, BooleanField, ReferenceField, ListField, IntField
+
+from brave.core.util.signal import update_modified_timestamp
+
+
+log = __import__('logging').getLogger(__name__)
+
+GRANT_WILDCARD = '*'
+
+
+class Permission(Document):
+    meta = dict(
+        collection='Permissions',
+        allow_inheritance = False,
+        indexes = [
+            dict(fields=['name'], unique=True, required=True)
+        ],
+    )
+    
+    name = StringField(db_field='n')
+    description = StringField(db_field='d')
+    
+    @property
+    def application(self):
+        """Returns the application that this permission is for."""
+        
+        from brave.core.application.model import Application
+        
+        app_short = self.name[:(self.name.index('.'))]
```

I'd prefer that it fail loudly rather than silently, since there should be no permissions without a dot (All permissions are required to start with {appName}.)